Trimble.Modus.Components/Controls/TMInput/TMInput.xaml.cs

```diff
@@ -0,0 +1,484 @@
+    {
+        if (bindable is TMInput tmInput)
+        {
+            tmInput.UpdateBorderColors(tmInput);
+        }
+    }
+
+    private static void OnTextChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is TMInput tmInput)
+        {
+            Tuple<bool, string> result = null;
+
+            if (!string.IsNullOrEmpty((string)newValue))
+            {
+                result = tmInput.InputValidation?.Invoke(tmInput);
+                if (result != null && !string.IsNullOrEmpty(result.Item2))
+                {
+                    tmInput.inputHelperLabel.Text = result.Item2;
+                    var response = result.Item1 ? ValidationResponse.Success : ValidationResponse.Error;
+                    // To avoid multiple times updating needs to check old value not equal to current value
+                    if (tmInput._validationResponse != response)
+                    {
+                        tmInput._validationResponse = response;
+                        SetBorderColor(tmInput);
+                    }
+
+                }
+            }
+            else
+            {
+                // To avoid multiple times updating needs to check old value not equal to current value
+                if (tmInput._validationResponse != ValidationResponse.Info)
+                {
+                    tmInput._validationResponse = ValidationResponse.Info;
+                    SetBorderColor(tmInput);
+                }
+            }
+            tmInput.TextChanged?.Invoke(tmInput, new TextChangedEventArgs((string)oldValue, (string)newValue));
+        }
+    }
+
+    private static void SetBorderColor(TMInput tmInput)
+    {
+        if (!tmInput.inputBorderlessEntry.IsFocused)
+        {
+            tmInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.Black);
+            tmInput.inputBorder.StrokeThickness = 1;
+
+            if (!string.IsNullOrEmpty(tmInput.HelperText))
+            {
+                tmInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.HelperImage);
+                tmInput.inputHelperLabel.Text = tmInput.HelperText;
+            }
+        }
+        else
+        {
+            switch (tmInput._validationResponse)
+            {
+                case ValidationResponse.Success:
+                    tmInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.Green);
+                    tmInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.TickImage);
+                    tmInput.inputBorder.StrokeThickness = 1;
+                    break;
+                case ValidationResponse.Error:
+                    tmInput.inputBorder.Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.DangerRed);
+                    tmInput.inputHelperIcon.Source = ImageSource.FromFile(ImageConstants.ErrorImage);
+                    tmInput.inputBorder.StrokeThickness = 1;
```

Comment:
Border thickness is 2 for valid and error state of the input

Reply:
Updated with 2